src/kv/rocksdb_cache/BinnedLRUCache.h

```diff
@@ -71,7 +71,13 @@ struct BinnedLRUHandle {
 
   uint32_t hash;     // Hash of key(); used for fast sharding and comparisons
 
-  char key_data[1];  // Beginning of key
+  char* key_data;  // Beginning of key
+
+  void SetKey(const char *src_data, size_t length) {
+    key_length = length;
+    key_data = new char[length];
```

Comment:
shall we call `delete[] key_data` before overwriting it?

Reply:
Yes, that's a good idea.  Either that, or we get rid of SetKey and move this back into BinnedLRUCacheShard::Insert where we know we just created the LRUHandle.  Any opinion?

Reply:
Ok, I decided to just ditch SetKey for a more minimal change in this PR.  Since we are now creating the LRUHandle and key_data in the same function I think we can avoid the extra delete.  I'd like to refactor this code at some point in the future but for now I think this is maybe the way to go.